src/integrationTest/java/uk/gov/hmcts/darts/arm/service/CleanupArmResponseFilesServiceIntTest.java

```diff
@@ -0,0 +1,677 @@
+import uk.gov.hmcts.darts.testutils.IntegrationBase;
+import uk.gov.hmcts.darts.testutils.data.MediaTestData;
+
+import java.time.LocalDate;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.darts.common.enums.ExternalLocationTypeEnum.ARM;
+import static uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum.ARM_RESPONSE_CHECKSUM_VERIFICATION_FAILED;
+import static uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum.ARM_RESPONSE_MANIFEST_FAILED;
+import static uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum.ARM_RESPONSE_PROCESSING_FAILED;
+import static uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum.STORED;
+
+@SpringBootTest
+@ActiveProfiles({"intTest", "h2db"})
+class CleanupArmResponseFilesServiceIntTest extends IntegrationBase {
+
+    private static final LocalDate HEARING_DATE = LocalDate.of(2023, 9, 26);
+
+    @Autowired
+    private ExternalObjectDirectoryRepository externalObjectDirectoryRepository;
+    @Autowired
+    private ObjectRecordStatusRepository objectRecordStatusRepository;
+    @Autowired
+    private ExternalLocationTypeRepository externalLocationTypeRepository;
+    @MockBean
+    private ArmDataManagementApi armDataManagementApi;
+    @MockBean
+    private UserIdentity userIdentity;
+    @MockBean
+    private ArmDataManagementConfiguration armDataManagementConfiguration;
+    @MockBean
+    private CurrentTimeHelper currentTimeHelper;
+
+    private CleanupArmResponseFilesService cleanupArmResponseFilesService;
+
+    @BeforeEach
+    void setupData() {
+
+        cleanupArmResponseFilesService = new CleanupArmResponseFilesServiceImpl(
+            externalObjectDirectoryRepository,
+            objectRecordStatusRepository,
+            externalLocationTypeRepository,
+            armDataManagementApi,
+            userIdentity,
+            armDataManagementConfiguration,
+            currentTimeHelper
+        );
+    }
+
+    @Test
+    void cleanupResponseFilesSuccessWithFiles_InputUpload_CreateRecord_UploadFile_AndStateStored() {
+
+        HearingEntity hearing = dartsDatabase.createHearing(
+            "Bristol",
+            "Int Test Courtroom 1",
+            "1",
+            HEARING_DATE
+        );
+
+        MediaEntity savedMedia = dartsDatabase.save(
+            MediaTestData.createMediaWith(
+                hearing.getCourtroom(),
+                OffsetDateTime.parse("2023-09-26T13:00:00Z"),
+                OffsetDateTime.parse("2023-09-26T13:45:00Z"),
+                1
+            ));
+        savedMedia.setChecksum("C3CCA7021CF79B42F245AF350601C284");
```

Comment:
I thought our checksums seemed to end with == e.g. `T2UrmSYWNZmvvYqBcMnV0g==`.
I think we were expecting them to be based on what we see in the Azure Blob Storage CONTENT-MD5 tag value as a base64 encoded representation of the binary MD5 hash value.
Hopefully we don't have a mismatch here of expected values for ARM checksum verification? What do you think @hemantasharma1129

Reply:
This was done when checksum was nullable but its now changed so has a value set when created. I have taken this out.